src/lxc/conf.c

```diff
@@ -3079,6 +3082,35 @@ static bool verify_start_hooks(struct lxc_conf *conf)
 	return true;
 }
 
+static int lxc_setup_console_ringbuf(struct lxc_console *console)
+{
+	int ret;
+	struct lxc_ringbuf *buf = &console->ringbuf;
+	uint64_t size = console->log_size;
+
+	if (size <= 0)
+		return 0;
+
+	ret = lxc_ringbuf_create(buf, size);
```

Comment:
Doesn't lxc_console_free() need to be updated?

Reply:
I guess it's just an mmap so no big deal

Reply:
That is an interesting question. I think not because `lxc_console_free()` is only ever called when you allocate the console or ttys from an already running container. In this case the console ringbuffer will have already been setup by the parent so when you detach from the console again but you'd free the ringbuffer you'd make the parent crash since the console epoll handler will still try to write to it.